centreon/src/Core/AgentConfiguration/Infrastructure/API/UpdateAgentConfiguration/UpdateAgentConfigurationController.php

```diff
@@ -0,0 +1,98 @@
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\AgentConfiguration\Infrastructure\API\UpdateAgentConfiguration;
+
+use Centreon\Application\Controller\AbstractController;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\AgentConfiguration\Application\UseCase\UpdateAgentConfiguration\UpdateAgentConfiguration;
+use Core\AgentConfiguration\Application\UseCase\UpdateAgentConfiguration\UpdateAgentConfigurationRequest;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Infrastructure\Common\Api\DefaultPresenter;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+final class UpdateAgentConfigurationController extends AbstractController
+{
+    use LoggerTrait;
+
+    public function __invoke(
+        int $id,
+        Request $request,
+        UpdateAgentConfiguration $useCase,
+        DefaultPresenter $presenter,
+    ): Response {
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+
+        try {
+            $updateAcRequest = $this->createRequest($id, $request);
+            $useCase($updateAcRequest, $presenter);
+        } catch (\InvalidArgumentException $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+            $presenter->setResponseStatus(new InvalidArgumentResponse($ex));
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+            $presenter->setResponseStatus(new ErrorResponse($ex));
+        }
+
+        return $presenter->show();
+    }
+
+    /**
+     * @param int $id
+     * @param Request $request
+     *
+     * @throws \InvalidArgumentException
+     *
+     * @return UpdateAgentConfigurationRequest
+     */
+    private function createRequest(int $id, Request $request): UpdateAgentConfigurationRequest
+    {
+        /**
+         * @var array{
+         *     name:string,
+         *     type:string,
+         *     poller_ids:int[],
+         *     configuration:array<string,mixed>
+         * } $data
+         */
+        $data = $this->validateAndRetrieveDataSent($request, __DIR__ . '/UpdateAgentConfigurationSchema.json');
+
+        $schemaFile = match ($data['type']) {
+            'telegraf' => 'TelegrafConfigurationSchema.json',
+            default => throw new \InvalidArgumentException(sprintf("Unknown parameter type with value '%s'", $data['type']))
+        };
+
+        $this->validateDataSent($request, __DIR__ . "/../Schema/{$schemaFile}");
+
+        $updateRequest = new UpdateAgentConfigurationRequest();
+        $updateRequest->id = $id;
+        $updateRequest->type = $data['type'];
+        $updateRequest->name = $data['name'];
+        $updateRequest->pollerIds = $data['poller_ids'];
+        $updateRequest->configuration = $data['configuration'];
```

Comment:
Are you sure you have these keys every time ? Maybe you should use isset function no ?

Reply:
it's ok !